test/src/folding/parsers/highlight_parser_test.dart

```diff
@@ -158,119 +142,314 @@ import java.util.Date;
+val a = "{[("
+val b = ")]}"''';
+      _parseAndCheck(mode: scala, code: code, expected: []);
+    });
+
+    test('Scala. Pair characters in comments are ignored.', () {
+      const code = '''
+class MyClass() {
+/// {[(                  1
+/*                       2
+  )]}                    3
+*/                       4
+/// {                    5
+}''';
+      const expected = [
+        _FB(startLine: 0, endLine: 6, type: _T.braces),
+        _FB(startLine: 2, endLine: 4, type: _T.multilineComment),
+      ];
+      _parseAndCheck(mode: scala, code: code, expected: expected);
+    });
+
+    test('Scala. Single line comments in a row.', () {
+      const code = '''
+// License                                            0
+// License                                            1
+class MyClass() {
+
+/// Method                                            4
+
+/// Method                                            6
+
+def method() = {} // Not the only thing in the line   8
+// Single                                             9
+def method2 // Not the only thing in the line        10
+// Single                                            11
+() = {}
+}
+// Single                                            14
+// Single                                            15''';
+      const expected = [
+        _FB(startLine: 0, endLine: 1, type: _T.singleLineComment),
+        _FB(startLine: 2, endLine: 13, type: _T.braces),
+        _FB(startLine: 4, endLine: 6, type: _T.singleLineComment),
+        _FB(startLine: 14, endLine: 15, type: _T.singleLineComment),
+      ];
+      _parseAndCheck(mode: scala, code: code, expected: expected);
+    });
+
+    test('Scala. Imports at the beginning.', () {
+      const code = '''
+package users
+/* */
+import users._  // import everything from the users package                      2
+/*                                                                               3
+  awesome multiline comment                                                      4
+*/
+import users.User  // import the class User                                      6
+import users.{User, UserPreferences}  // Only imports selected members           7
+import users.{UserPreferences => UPrefs} // import and rename for convenience    8
+
+class MyClass() {} //                                                           10 ''';
+      const expected = [
+        _FB(startLine: 0, endLine: 2, type: _T.imports),
+        _FB(startLine: 3, endLine: 5, type: _T.multilineComment),
+        _FB(startLine: 6, endLine: 8, type: _T.imports),
+      ];
+      _parseAndCheck(mode: scala, code: code, expected: expected);
+    });
+
+    test('Scala. Imports at mid and the end, no newline in the end.', () {
+      const code = '''
+//
+package users
+import users._  // import everything from the users package
+class MyClass() {}
+import users.User  // import the class User
+import users.{User, UserPreferences}  // Only imports selected members''';
```

Comment:
What happens if we break this line?
```scala
import users.{
  User,
  UserPreferences
}
```

Reply:
Created an issue: https://github.com/akvelon/flutter-code-editor/issues/78